libvast/src/system/archive.cpp

```diff
@@ -185,9 +185,17 @@ archive(archive_type::stateful_pointer<archive_state> self, path dir,
           t.stop(events);
         },
         [=](unit_t&, const error& err) {
-          if (err) {
-            VAST_ERROR(self, "got a stream error:", self->system().render(err));
+          // We get an 'unreachable' error when the stream becomes unreachable
+          // because the actor was destroyed; in this case we can't use `self`
+          // anymore.
```

Comment:
In retrospect I'm not sure anymore that the comment is 100% accurate: Removing the check in the partition gives occasional segfaults due to null pointer dereference of `self`, but a trivial reproduction of the situation did not have the same issue, so maybe this is something specific to the setup of the index/partition. (i was planning to investigate this for the streaming blog post as well, but didn't really get to it yet)

Reply:
Should we investigate this separately from this PR?